expression/scalar_function.go

```diff
@@ -125,6 +125,40 @@ func (sf *ScalarFunction) MarshalJSON() ([]byte, error) {
 	return []byte(fmt.Sprintf("\"%s\"", sf)), nil
 }
 
+// typeInferForNull infers the NULL datums field type and set the field type
+// of NULL datum same as other non-null operands.
+func typeInferForNull(args []Expression) {
```

Comment:
Why do we need this operation?

Reply:
We are planning to remove the old computing framework in the Coprocessor and use the vectorize computing framework to handle all coprocessor requests.  This BUG is detected by https://github.com/tikv/tikv/pull/5817. 

There are some reasons why I think this is a BUG:
1. NULL datum should have a type if it is involved in calculation with non-NULL instead of `mysql.TypeNull`. (e.g: `1 + NULL`, the `NULL` should be a `mysql.TypeLong`)
2. The `mysql.TypeNull` exist because of `NULL + NULL` cannot infer the actual type.
3. NULL datum doesn't mean it should be `mysql.TypeNull`.